parse/parse.go

```diff
@@ -41,10 +48,16 @@ func GoSum(path string) (deps types.ProjectList, err error) {
 
 	scanner := bufio.NewScanner(file)
 	for scanner.Scan() {
-		s := strings.Split(scanner.Text(), " ")
-		if !strings.HasSuffix(s[1], "/go.mod") {
-			deps.Projects = append(deps.Projects, types.Projects{Name: s[0], Version: s[1]})
-		}
+		parseGoModFormattedDependency(scanner, &deps)
 	}
 	return deps, nil
 }
+
+// Parses a line of text that is in go.sum/go list -m all format. Which is basically just `<dep name> <version number>`
+func parseGoModFormattedDependency(scanner *bufio.Scanner, deps *types.ProjectList) {
+	text := scanner.Text()
+	s := strings.Split(text, " ")
+	if len(s) > 1 && !strings.HasSuffix(s[1], "/go.mod") {
```

Comment:
I think we may not need the check on `/go.mod`, I don't believe I see that in any of the pipe from `go list -m all`. Do you want to verify @zendern ?

Reply:
I'll double check but i know i re-used this method for go list and go.mod parsing so thats why its there.

Reply:
Ill double check but i use that method in both scenarios since once you get to the dependency parsing its the same.

Reply:
Super fair point, I hadn't thought about that. Buyers choice on if you do anything about it as a result :)

Reply:
I'll still take a look. Maybe we can remove it or maybe I duplicate a little code just for better clarity there.

Reply:
Alright tweaked it a little to hopefully make it clearer that `go list -m all`  dependency parsing doesn't really need any checks for `go.mod` anything. Let me know your thoughts on that change.